### PR TITLE
EAHW 1249 (annual target hours)

### PR DESCRIPTION
**Pull request by wweatherill-homeoffice**

### CODE REVIEW

**Review by dawatkin3 — CHANGES_REQUESTED** [review]
Small typo on Allowence - Allowance

**Comment by wweatherill-homeoffice** [comment]
> Small typo on Allowence - Allowance

Should be addressed in the latest commit @dawatkin3

**Review by dawatkin3 — CHANGES_REQUESTED** [review]
Unless there is a good reason would prefer we retained FteValue rather than is_fulltime
Don't understand what terms_and_conditions_id is.
location_id - since it comes from METIS is likely to be a string.

**Comment by wweatherill-homeoffice** [comment]
> Unless there is a good reason would prefer we retained FteValue rather than is_fulltime Don't understand what terms_and_conditions_id is. location_id - since it comes from METIS is likely to be a string.

Replaced is_fulltime with fte_value
terms_and_conditions_id - dropped. Left in by mistake
location_id I thought from a DB perspective it's type would be VARCHAR

**Review by dawatkin3 — DISMISSED** [review]

File: docs/commands.md [hunk 1]
```diff
@@ -50,6 +51,21 @@ This command retrieves a set of `TimeEntry` instances whose time period (time co
  - business failure - see [Record Time](https://collaboration.homeoffice.gov.uk/jira/browse/EAHW-925) (access required) for business failure scenarios. Also see [standard command output](TODO) (**TODO**)  for how to report business failures
  - technical failure - see [standard command output](TODO) (**TODO**) for how to report technical failures
 
+## add FlexChange
+This command causes a new `FlexChange` to be created. The creation of a `FlexChange` is driven by a change to an existing `TimeEntry` (change to the `TimeEntry.actualStartTime`and/or the `TimeEntry.actualEndTime` values). Without this modification the `FlexChange` cannot exist. On this basis it is crucial that the implementation of this command ensures that the `TimeEntry` modification and the `FlexChange` creation are committed as an atomic unit. If either operation fails then the whole command should fail and any persisted changes should be rolled back to the state that the entities were in prior to the command being invoked.
```
Comment (laurieboyes) [inline docs/commands.md:55]: Does`TimeEntry.actualStartTime` here want to be updated to reflect the new field names for the times?

File: docs/commands.md [hunk 2]
```diff
@@ -81,4 +97,21 @@ This command will ultimately retrieve a filtered list of TimeEntryTypes appropri
...
+
+### inputs 
+- TenantId    - mandatory -  the tenant whose `FlexChangeType` instances will be searched
+
+### output
+ - success - see [standard command output](TODO)(**TODO**)  for how to report success output
+ - business failure - see [standard command output](TODO) (**TODO**) for how to report technical failures
+ - response payload - upon a successful call to get FlexChangeType a payload containing a list of all relevant FlexChangeType must be returned.
  
+ ### FlexChangeType list
```
Comment (laurieboyes) [inline docs/commands.md:112]: I think you lose your heading formatting with this wayward whitespace
```suggestion
### FlexChangeType list
```

File: docs/features/eahw-1249-annual-target-hours.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# EAHW-1249 Annual Target Hours
+
+This high-level design is intended to cover the elements of the Annual Target Hours feature that effect TimeCard. It seeks to:
```
Comment (laurieboyes) [inline docs/features/eahw-1249-annual-target-hours.md:3]: Typo
```suggestion
This high-level design is intended to cover the elements of the Annual Target Hours feature that affect TimeCard. It seeks to:
```

File: docs/features/eahw-1249-annual-target-hours.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+When a `TimeEntry` is successfully recorded by the TimeCard container it triggers the publication of an event that encapsulates that `TimeEntry` resource along with the type of action that was performed on the resource. As an event producer the TimeCard container should follow the blueprints below - 
+
+### Topic
+Based on the guideance in the [topic creation](https://github.com/UKHomeOffice/callisto-docs/blob/main/blueprints/topic-creation.md) blueprint the following are suggested - 
+
+- name - callisto-timecard
+- partition key - `TimeEntry.ownerId`
+
+### Event
+Based on the guideance in the [event publication (schema & trigger points)](https://github.com/UKHomeOffice/callisto-docs/blob/main/blueprints/event-publishing-and-consuming.md) blueprint the following are suggested -
```
Comment (laurieboyes) [inline docs/features/eahw-1249-annual-target-hours.md:25]: typo
```suggestion
Based on the guidance in the [event publication (schema & trigger points)](https://github.com/UKHomeOffice/callisto-docs/blob/main/blueprints/event-publishing-and-consuming.md) blueprint the following are suggested - 
```

File: docs/features/eahw-1249-annual-target-hours.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+This design builds upon the design for [Record Time](./record-time.md) and introduces some changes to that design (see below)
+
+The Annual Target Hours feature primarily impacts the Accruals container however the process detailed in that feature is triggered by events published by the TimeCard container.
+
+## Considerations
+When a `TimeEntry` is successfully recorded by the TimeCard container it triggers the publication of an event that encapsulates that `TimeEntry` resource along with the type of action that was performed on the resource. As an event producer the TimeCard container should follow the blueprints below - 
+
+### Topic
+Based on the guideance in the [topic creation](https://github.com/UKHomeOffice/callisto-docs/blob/main/blueprints/topic-creation.md) blueprint the following are suggested -
```
Comment (laurieboyes) [inline docs/features/eahw-1249-annual-target-hours.md:19]: typo
```suggestion
Based on the guidance in the [topic creation](https://github.com/UKHomeOffice/callisto-docs/blob/main/blueprints/topic-creation.md) blueprint the following are suggested - 
```

File: docs/container-model.c4 [hunk 6]
```diff
@@ -92,6 +92,40 @@ workspace {
 			timecardService -> singlePageApplication "return success or failure"
```
Comment (laurieboyes) [inline docs/container-model.c4:92]: Is there a way to review c4 files without trying to read the markup?

**Review by laurieboyes — APPROVED** [review]

File: docs/container-model.c4 (hunk 6 above)
Comment (wweatherill-homeoffice) [reply docs/container-model.c4:92]: Copy and paste C4 file content into https://structurizr.com/dsl

**Review by taomoh — APPROVED** [review]

**Review by stephenkaye-ho — APPROVED** [review]

**Pull request merged**